packages/extensionmanager/src/listings.ts

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { ISignal, Signal } from '@lumino/signaling';
+
+import { URLExt } from '@jupyterlab/coreutils';
+
+import { ServerConnection } from '@jupyterlab/services';
+
+/***
+ * Information about a listed entry.
+ */
+export interface IListEntry {
+  /**
+   * The name of the extension.
+   */
+  name: string;
+  regexp: RegExp | undefined;
+  type: string;
+  reason: string;
+  creation_date: string;
+  last_update_date: string;
+}
+
+/**
+ * Listing search result structure (subset).
+ *
+ */
+export interface IListResult {
```

Comment:
Think this might be better suited as a type union, since you can only have a blacklist or whitelist, not both. Something like this:

```typescript
type ListResult = null | {mode: 'white' | 'black', uris: string[], entries: IListEntry[] }
```

Reply:
Makes sense. Updates pushed with your proposal.